### ISSUE TITLE
Access to server error response body on processfile

### ISSUE THREAD

**Issue by leoliu** [pqina/filepond#112]
I have replaced the upload UI of an existing project with filepond and loved it. I have a request regarding error handling.

When an upload failed the filepond UI has generic error indicator which often doesn't tell much what to do to fix them if fixable. My server may send more informative error messages and I like to get at them handily. Here is what I am doing now:

```js
    this.pond.on("processfile", (error, file) => {
      // XXX: Cannot get the error the server returns
      error
        ? toast({ type: "error", text: error.code + " " + error.body })
        : this.uploadSucceeded(file);
    });
```

which shows a notification with generic HTTP error messages. Should there be a way to access the response body? There is an onerror hook on the `server` configuration option to get just the body. But combining these two hooks (processfile and onerror) is a bit of a hassle.

**Comment by rikschennink** [comment]
The `onerror` handler on the `server.process` property allows you to select the error message from the server response to show in the file item. Is that not what you're looking for?

**Comment by leoliu** [comment]
@rikschennink,

Thanks for the pointer. I'll finally get the hang of what onerror does that satisfy my needs for now. But if custom error messages can show up on the file items it is much better usability-wise.

Thanks.

**Comment by rikschennink** [comment]
Hi @leoliu just published version `2.1.0` it adds the option to set a function to a label property. The function receives the current error object, based on that you can then decide what to show the user.

```js
FilePond.create({
    labelFileProcessingError: (error) => {
        console.log(error);
        return 'My server errored';
    }
}
```

**Comment by leoliu** [comment]
Hi @rikschennink, I am using 2.1.0 and this new feature is neat. Thanks.
